### PR TITLE
google-oauth

### PR DESCRIPTION
**Pull request by ibez92**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,3017 @@
+{
```
Comment (flungloaf) [inline package-lock.json]: We are using `pnpm` here, please remove this.

File: utils/auth/client.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ interface SignInResponse {
 interface AuthClient {
   signIn: (email: string, password: string) => Promise<SignInResponse>;
   signUp: (email: string, password: string) => Promise<SignInResponse>;
+  oAuthGoogleSignin: (token: string) => Promise<SignInResponse>;
```
Comment (flungloaf) [inline utils/auth/client.ts]:
```suggestion
  googleSignin: (token: string) => Promise<SignInResponse>;
```

File: utils/auth/provider.tsx [hunk 3]
```diff
@@ -62,12 +72,15 @@ const AuthProvider = ({ children }: PropsWithChildren) => {
...
         signup,
         signout,
+        oAuthGoogleSignin,
         accessToken,
         refreshToken,
         authenticated,
       }}
     >
-      {children}
+      <GoogleOAuthProvider clientId="43480745913-1lhqt43oe2gir7fqdv5058m5l3q1hl2p.apps.googleusercontent.com">
```
Comment (flungloaf) [inline utils/auth/provider.tsx]: Let's add this as a `.env` config variable, read the guide [here](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser)

**Review by flungloaf — CHANGES_REQUESTED** [review]

File: utils/auth/provider.tsx (hunk 3 above)
Comment (ibez92) [reply utils/auth/provider.tsx]: done

File: utils/auth/client.ts (hunk 2 above)
Comment (ibez92) [reply utils/auth/client.ts]: fixed

File: package-lock.json (hunk 1 above)
Comment (ibez92) [reply package-lock.json]: fixed

**Pull request merged**